### DISCUSSION TITLE
Is it possible to omit config file in workspace projects?

### DISCUSSION THREAD

**Discussion by wctiger** [vitest-dev/vitest#6194] [Q&A] [answered]
**Background:** 

Assume there's a monorepo with 100 projects built with nx and pnpm workspace. And also assume these projects in the monorepo can share the exact same vitest config. So it will be really nice if all these projects can be sharing the same config file. This can be achieved by having a vitest.config.ts in each project in a workspace like

```
// vitest.workspace.ts
import { defineWorkspace } from 'vitest/config';

export default defineWorkspace([
  'packages/*',
]);


// vitest.shared.ts
import { defineProject } from 'vitest/config'

export default defineProject({
  resolve: {
    '@': resolve(__dirname, 'src')
  }
  test: {
    environment: 'jsdom',
  }
})


// packages/a/vitest.config.ts
import configShared from '../../vitest.shared'

export default configShared;
```

However this is not ideal as I will need to copy pasting the same `vitest.config.ts` in all projects. So I was trying to see if this shared config can be directly applied to all workspace projects without even having the `vitest.config.ts`. 

Another concern here is to make the [VSCode extension](https://github.com/vitest-dev/vscode) to work in this workspace which is why I was also keen to stick to the Vitest Workspace setup. Otherwise I can just have a CLI command run by nx pointed directly to that shared config, but this setup doesn't seem to be liked up by the VSCode extension.

Any idea is appreciated!

**Comment by AriPerkkio** [comment, marked as answer]
Something like this could work:
- Remove `vitest.shared.ts`
- Remove all `packages/*/vitest.config.ts`
- Modify `vitest.workspaces.ts` with following:

```ts
// vitest.workspace.ts
import { readdirSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { defineProject, defineWorkspace } from 'vitest/config'

const projects = readdirSync("./packages");

export default defineWorkspace(
  projects.map((project) => defineProject({
    test: {
      name: project,
      root: `packages/${project}`,
      environment: "jsdom",
      alias: {
        "@": fileURLToPath(new URL(`packages/${project}/src`, import.meta.url)),
      },
    },
  }))
);
```

Example: https://stackblitz.com/~/edit/vitest-sonar-reporter-workspaces-gbhuh4?file=vitest.workspace.ts&view=editor
